### ISSUE TITLE
Variable Fallback breaks with at @ sign

### ISSUE THREAD

**Issue by aconz2** [serverless/serverless#5449]
# This is a Bug Report

## Description

* What went wrong?
Variable substitution breaks with an `@` symbol in a string fallback

* What did you expect should have happened?
All environment variable substitutions should be `foo@bar`
Additionally, conflicting reports on whether string fallbacks are supported, as the main documentation [site](https://serverless.com/framework/docs/providers/aws/guide/variables#syntax) demonstrates their use as `${opt:stage, 'dev'}`. But the closing comment from #2782 says they are only supported from other variables.

* What was the config you used?
```
service: sls-bug

custom:
  defaults:
    email: 'foo@bar'
  a: 'foo@bar'
  b: ${env:b, self:custom.defaults.email}  # the workaround from #2782
  c: ${env:c, "foo@bar"}
  d: ${env:d, 'foo@bar'}

provider:
  name: aws
  runtime: nodejs8.10

functions:
  hello:
    handler: handler.hello

    environment:
      a: ${self:custom.a}
      b: ${self:custom.b}
      c: ${self:custom.c}
      d: ${self:custom.d}

```
* What stacktrace or error message from your provider did you see?
```
→ sls package && jq '.Resources.HelloLambdaFunction.Properties.Environment.Variables'  .serverless/cloudformation-template-update-stack.json
Serverless: Packaging service...
Serverless: Excluding development dependencies...
{
  "a": "foo@bar",
  "b": "foo@bar",
  "c": "${env:c, \"foo@bar\"}",
  "d": "${env:d, 'foo@bar'}"
}
```
Similar or dependent issues:
* #2782
This was my inciting issue: the `@` symbol in a string fallback does not work and causes issue above
Quoted from comment closing issue:
"The issue is that you're trying to overwrite with string fallbacks, while our implementation only supports valid variable sources (ie. env:, self:, file:...etc)"
* #5398

## Additional Data

* ***Serverless Framework Version you're using***: 1.32.0
* ***Operating System***: Linux Fedora 28 4.18.16-100.fc27.x86_64
* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by exoego** [comment]
It seems already fixed in sls v1.36.1 in Ubuntu 18.04.1.

`c` and `d` are also populated correctly.
```
$  sls package && jq '.Resources.HelloLambdaFunction.Properties.Environment.Variables'  .serverless/cloudformation-template-update-stack.json
Serverless: Packaging service...
Serverless: Excluding development dependencies...
{
  "a": "foo@bar",
  "b": "foo@bar",
  "c": "foo@bar",
  "d": "foo@bar"
}
```

**Comment by pmuens** [comment]
Thanks for opening @aconz2 👍

Thanks for confirming @exoego 👍. Yes, this should've been fixed in the meantime.
